File: C/Algorithms/125.c
#include <stdbool.h>

bool isPalindrome(char * s){
    int start = 0, end = strlen(s) - 1;
    
    while (start < end) {
        while ((start < end) && !isalnum(s[start]))
            start++;
        while ((start < end) && !isalnum(s[end]))
            end--;
        if (tolower(s[start++]) != tolower(s[end--])) {
            return false;
        }
    }
    return true;    
}

File: Go/Algorithms/125.go
package main

import "strings"

func isLetter(c rune) bool {
	return ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z') || ('0' <= c && c <= '9')
}

func isPalindrome(s string) bool {
	var r strings.Builder
	for _, ch := range strings.ToUpper(s) {
		if isLetter(ch) {
			r.WriteRune(ch)
		}
	}
	s = r.String()
	start, end := 0, len(s)-1
	for ; start < end; start, end = start+1, end-1 {
		if s[start] != s[end] {
			return false
		}
	}
	return true
}
